src/components/Redemptions/RedemptionsPage.tsx

```diff
@@ -0,0 +1,269 @@
+          if (externalTokenRewards[contributionReward.externalToken]) {
+            externalTokenRewards[contributionReward.externalToken].iadd(new BN(contributionReward.externalTokenReward));
+          } else {
+            externalTokenRewards[contributionReward.externalToken] = new BN(contributionReward.externalTokenReward);
+          }
+        }
+      }
+
+    });
+
+    const totalRewards: any[] = [];
+    if (!ethReward.isZero()) {
+      totalRewards.push(formatTokens(ethReward, "ETH"));
+    }
+    if (!genReward.isZero()) {
+      totalRewards.push(formatTokens(genReward, "GEN"));
+    }
+    Object.keys(externalTokenRewards).forEach((tokenAddress) => {
+      totalRewards.push(formatTokens(externalTokenRewards[tokenAddress], tokenSymbol(tokenAddress)));
+    });
+    if (reputationRewardDaos.size > 0) {
+      totalRewards.push(<span>reputation in {reputationRewardDaos.size}&nbsp;DAOs</span>);
+    }
+
+    return <strong>
+      {totalRewards.reduce((acc, v) => {
+        return acc === null ? <React.Fragment>{v}</React.Fragment> : <React.Fragment>{acc} <em>&amp;</em> {v}</React.Fragment>;
+      }, null)}
+    </strong>;
+  }
+}
+
+const SubscribedRedemptionsPage = withSubscription({
+  wrappedComponent: RedemptionsPage,
+  loadingComponent: <div className={css.loading}><Loading/></div>,
+  errorComponent: (props) => <div>{ props.error.message }</div>,
+  checkForUpdate: ["currentAccountAddress"],
+  createObservable: (props: IStateProps) => {
+    const { currentAccountAddress } = props;
+
+    if (!currentAccountAddress) {
+      return of(null);
+    }
+
+    const arc = getArc();
+    const query = gql`
+      {
+        proposals(
+          where: {
+            accountsWithUnclaimedRewards_contains: ["${currentAccountAddress}"]
+          },
+          orderBy: closingAt
+        ) {
+          id
+          dao {
+            id
+          }
+          gpRewards {
+            beneficiary
+            tokensForStaker
+            tokensForStakerRedeemedAt
+            daoBountyForStaker
+            daoBountyForStakerRedeemedAt
+            reputationForVoter
+            reputationForVoterRedeemedAt
+            reputationForProposer
+            reputationForProposerRedeemedAt
+          }
+          contributionReward {
+            id
+            beneficiary
+            ethReward
+            externalToken
+            externalTokenReward
+            reputationReward
+          }
+        }
+      }
+    `;
+    const proposals = arc.getObservable(query)
+      .pipe(map((result: any) => result.data.proposals));
+    const daos = proposals
+      .pipe(mergeMap((proposals: any[]): Observable<IDAOState[]> => {
+        const daoIds = new Set(proposals.map((proposal: any) => { return proposal.dao.id; }));
+        return combineLatest(Array.from(daoIds, (id: any) => arc.dao(id).state()))
+          .pipe(defaultIfEmpty([]));
+      }));
+    return combineLatest(daos, proposals);
```

Comment:
is there any reason we actually need to to subscribe to the daos? i bet @jellegerbrandy has something to say about how we can reduce the number of subscriptions here given all the work he has been doing on that.

Reply:
The proposal cards on the page require a DAO state as a prop. It looks like various subcomponents use it: ProposalData, ActionButton, VoteButtons, StakeButtons, AccountPopup, etc.